apis/python/src/tiledbsoma/_flags.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2024 The Chan Zuckerberg Initiative Foundation
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Licensed under the MIT License.
+
+import os
+
+# Temporary for https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+_new_shape_feature_flag = os.getenv("SOMA_PY_NEW_SHAPE") is not None
+
+
+def _new_shape_feature_flag_enabled() -> bool:
+    """
+    This is temporary only and will be removed once
+    https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+    is complete.
+    """
+    return _new_shape_feature_flag
```

Comment:
What's the motivation behind having this live in a function? I would make `NEW_SHAPE_FEATURE_FLAG_ENABLED` be a public constant and use that.

Reply:
@thetorpedodog sure, whatever you like.

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/3005/commits/a8b5909bdc0d6c037454eb389a7afbe318c92a3d